static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -0,0 +1,138 @@
+                        component.templateRenderInfoRef = component.db.object('template-render-info/' + params.id);
+                        component.templateDirectoryInfo = component.templateDirectoryInfoRef.valueChanges();
+                        component.templateRenderInfo = component.templateRenderInfoRef.valueChanges();
+                        component.templateDirectoryInfoRef.valueChanges().subscribe((response) => {
+                            // component.templateDirectoryInfo = response;
+                            // metadataIsValid = false;
+                        });
+                        component.templateRenderInfoRef.valueChanges().subscribe((response) => {
+                            // component.templateRenderInfo = response;
+                            // metadataIsValid = false;
+                        });
+                        component.templateVariableNameList = component.db.object('template-render-info/' + params.id + '/variables')
+                            .valueChanges();
+                        // .subscribe((response) => {
+                        //     component.templateVariableNameList = response;
+                        //     if (!component.templateVariableNameList || !metadataIsValid) {
+                        //         component.router.navigate(['home']);
+                        //     }
+                        // });
+                    });
+                });
+            }
+        });
+    }
+
+    goHome() {
+        this.router.navigate(['']);
+    }
+
+    logout(): void {
+        this.authService.logout(null);
+    }
+
+    downloadTemplate() {
+        const component = this;
+        component.validateEnteredVariables();
+        // component.templateRenderInfoStorageRef.getDownloadURL().subscribe(url => console.log('targetUrl: ', url));
+        // console.log('targetUrl: ', this.targetUrl);
+
+        component.templateRenderInfoRef.snapshotChanges().subscribe(data => {
+            // console.log(data.payload.val().templateArchiveUrl);
+            const fileEndings = data.payload.val().fileEndings;
+            for (let i = 0; i < data.payload.val().fileEndings.length; i++) {
+                fileEndings[i] = fileEndings[i].name;
+            }
+            // console.log(component.templateRenderInfoStorageRef.getDownloadURL());
+            const request = encodeURIComponent(JSON.stringify({
+                'variables': component.valueMap,
+                'fileEndings': fileEndings,
+                'url': encodeURI(data.payload.val().templateArchiveUrl)
+            }));
+            console.log('Sending request: ', request);
+            const options = {responseType: 'blob' as 'blob'};
+            const linkElement = document.createElement('a');
+            component.http.get('http://localhost:3000/api/download-template?request=' + request, options)
```

Comment:
localhost should not be hardcoded - this is bad practice and will break in production. This is why all of this code should be moved into the api interface service - it exists to prevent these kinds of errors.